DesignPathSearch_II/InputDesignFilePaths.cs

```
@@ -0,0 +1,74 @@
+﻿using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace DesignPathSearch
+{
+    class InputDesignFilePaths
+    {
+
+        private List<DesignFilePath> InputDesignFilePathsList = new List<DesignFilePath>();
+
+        public InputDesignFilePaths() { }
+
+        public List<DesignFilePath> GetInputDesignFilePaths(string xtplDir, List<string> dgList)
+        {
+            string[] filePaths = null;
+            try
+            {
+                filePaths = Directory.GetFiles(xtplDir, "*.xtpl", SearchOption.AllDirectories);
+            }
+            catch (Exception ex)
```

Should this be an IOException?

---

Well, you raise an interesting question which I will try to answer.

According to the Microsoft C# Dcoumentation for .Net 4.6.1  the following exceptions can be raised for Directory.GetFile method ...

IOException
UnauthorizedAccessException
AurguementException
ArguementNullException
PathTooLongException
DirectoryNotFoundException

Thus if I want to catch a log all possible exceptions at the point they occur then Microsoft recommends catch (Exception e) which will catch all exceptions throw in the try block.

Thus is the coding standard I followed.

If I was going to handle one or more of the possible exceptions differently in code, then i code add addition catch blocks ....

Thus 

catch (IOException ioe)
{
     // do something
}
catch (Exception ex)
{
   // do something
}

In the case of this block of code, I choose to to contiinue, the thus logging all exceptions was I considered the best approach.

---

Fair enough - you could just handle IOException and Exception, since some of the exceptions types listed above inherit from IOException - that seems like a clean enough distinction between the possible error types that might be thrown. But yeah, since you're just logging in any case, it's probably not tremendously important.